internal/controller/pipelinerollout/pipelinerollout_progressive.go

```diff
@@ -108,54 +119,54 @@ func (r *PipelineRolloutReconciler) AssessUpgradingChild(
 	if err != nil {
 		return assessment, reasonFailure, err
 	}
+
 	// if we fail even once, that's considered failure
 	if assessment == apiv1.AssessmentResultFailure {
 		// set AssessmentEndTime to now and return failure
+		numaLogger.Debugf("Assessment failed for upgrading child %s", existingUpgradingChildDef.GetName())
 		_ = progressive.UpdateUpgradingChildStatus(pipelineRollout, func(status *apiv1.UpgradingChildStatus) {
-			assessmentEndTime := metav1.NewTime(time.Now())
-			status.BasicAssessmentEndTime = &assessmentEndTime
+			status.TrialWindowStartTime = nil
+			status.AssessmentResult = apiv1.AssessmentResultFailure
```

Comment:
I think this should return AssessmentResultUnknown. We don't want to return failure here.